group/assignment6/RearrrangingCars.java

```diff
@@ -0,0 +1,71 @@
+package assignment_6;
+
+import java.util.Hashtable;
+import java.util.LinkedList;
+import java.util.Map.Entry;
+import java.util.Random;
+
+public class RearrrangingCars {
+
+	private Hashtable<Integer, Character> start;
+	private Hashtable<Integer, Character> end;
+	private Random rand;
+	private int noCarKey = 0;
+	private int noExistKey = -1;
+
+	public RearrrangingCars(Hashtable<Integer, Character> start,
+			Hashtable<Integer, Character> end) {
+		rand = new Random();
+		this.start = start;
+		this.end = end;
+	}
+	
+	/**
+	 * given an starting and ending state of parking
+	 * start with the parking emptySlot in start state,
+	 * get the car that will exist at the same slot in the end state 
+	 * and do a move by swap the car from its current slot to the emptySlot
+	 * repeat that until reaching emptySlots
+	 * finally check if the start state equal the end,
+	 * if not it will be the case that at the beginning the emptySlot at the same position in two states
+	 * in this case generate in the first move a random slot to swap it with the empty 
+	 * and then starting the algorithm
+	 * @return linkedList of sequences to define moves as car, starting and ending spaces
+	 */
+	public LinkedList<Sequence> generateMoves() {
+		LinkedList<Sequence> moves = new LinkedList<>();
+		char emptySlot = start.get(noCarKey);
+		move(moves, noCarKey, emptySlot);
+		if (!start.equals(end)) {
+			int key = rand.nextInt(start.keySet().size() - 1) + 1;

```

Comment:
Here we need to check that the random key does not correspond to a car that was already in the right place or to the empty slot!

Reply:
yes, i am working on it now